hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -510,6 +513,15 @@ BalanceProcedureScheduler getFedRenameScheduler() {
     return this.fedRenameScheduler;
   }
 
+  /**
+   * Get the routerStateIdContext used by this server.
+   * @return routerStateIdContext
+   */
+  @VisibleForTesting
+  public RouterStateIdContext getRouterStateIdContext() {
```

Comment:
Does protected not work? i.e. Your test should be able to access a protected member.

Reply:
Protected works too. If we need to increase visibility later, we can do so.